code/game/machinery/ARES/ARES_procs.dm

```diff
@@ -0,0 +1,97 @@
+GLOBAL_DATUM_INIT(ares_link, /datum/ares_link, new)
+
+/datum/ares_link
+	var/link_id = MAIN_SHIP_DEFAULT_NAME
+	var/obj/structure/machinery/ares/processor/interface/p_interface
+	var/obj/structure/machinery/ares/processor/apollo/p_apollo
+	var/obj/structure/machinery/ares/processor/bioscan/p_bioscan
+	var/obj/structure/machinery/computer/ares_console/interface
```

Comment:
null these out on Destroy()

In addition, make these null in their respective Destroy()s

Reply:
alternatively, make these weakrefs